portfolio/src/main/webapp/script.js

```diff
@@ -16,10 +16,51 @@
  * Adds a random fun fact to the page.
  */
 
-// Spotify API keys to be used for later
-// var client_id = '444dda318d8949cb9282d95a9885a7c6';
-// var client_secret = '9f81ba6cf5748ef9d24a7ddb1e8bc4c';
-// var redirect_uri = 'https://8080-dot-12317386-dot-devshell.appspot.com/misc.html';
+function formatAsMinutesAndSeconds(milliseconds) {
+  const minutes = Math.floor(milliseconds / 60000);
+  const seconds = ((milliseconds % 60000) / 1000).toFixed(0)
+  return minutes + ':' + (seconds < 10 ? '0' : '') + seconds;
+}
+
+function getCurrentlyPlayingSong() {
+  $.get("./auth.txt", function(data) {
```

Comment:
This is risky too, because it means anybody could visit ./auth.txt and get your oauth token, which they can then use to impersonate you on Spotify. In order for this to work securely you also have to move the 'currently_playing' API request to the server, so that anybody in the world can check what you're currently listening to but can't do anything else with the Spotify API